can/map/map.js

```diff
@@ -57,13 +61,13 @@ var canMapBehavior = behavior("can/map",function(baseConnection){
 				}
 			}
 
-			this.List = this.List || this.Map.List;
+			this.List = this.List || this.ArrayType || this.Map.List;
```

Comment:
I’m not sure if we’d also want to support `this.ObjectType.ArrayType`

Reply:
Let's find out if there are any guides that would use this.

Reply:
I just did a quick search. It looks like the majority of our guides show setting `Map.List` but pass both `Map` and `List` options. It looks like only the 5.x migration guide shows passing `Map` but no `List` option as well.

Reply:
Yeah, but in 6.x we've been changing these because this is ugly:

```js
class Message extends ObservableObject { ... }

Message.ArrayType = class MessageList extends ObservableArray { ... }
```

I think the codemods might do this automatically? cc @Mattchewone

Reply:
cc @phillipskevin Yeah I had it transform `Users.List = DefineList.extend(...)` into `class UserListModel extends ObservableArray { ... }`